src/luapatt.py

```diff
@@ -443,9 +444,12 @@ def matchclass(self, sc, pc):
             match = sc.isdigit() or sc in 'abcdefABCDEF'
         elif pcl == 'z':
             match = sc == '\0'
-        elif pcl in 'cgp':
-            raise NotImplementedError('{0}c, {0}g, and {0}p are not '
-                                      'available'.format(ESCAPE))
+        elif pcl == 'c':
+            match = unicodedata.category(sc)[0] == 'C'
+        elif pcl == 'g':
+            match = unicodedata.category(sc)[0] not in ('C', 'Z')
```

Comment:
I don't like tuples of single characters. Please change the tuple to a string (like in the removed line 446).

```suggestion
            match = unicodedata.category(sc)[0] not in 'CZ'
```